students/merrix/session06/mailroom4.py

```diff
@@ -0,0 +1,102 @@
+import sys  # imports go at the top of the file
+import math 
+
+from collections import defaultdict
+
+donor_db = defaultdict(list)
+
+
+donor_db = {"William Gates":[653772, 12.17],
+            "Jeff Bezos": [877.33],
+            "Paul Allen" : [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0]
+           }
+
+
+prompt = "\n".join(("Welcome to James Dot Donations!",
+          "Please choose from below options:",
+          "1 - Send a Thank you",
+          "2 - Create a report",
+          "3 - Send Letter",
+          "4 - To Exit",
+          ">>> "))
+
+
+def view_donor_db():
+    print (donor_db.keys())
+    choose = input("enter a name from the list or a new name to donate: ").title()
+    if choose not in donor_db.keys():
+        add_donor_db(choose)
+    elif choose in donor_db.keys():
+        donation = int(input("Enter an amount to donate to %s " % choose)) 
+        donor_db[choose].append(int(donation))
+        print (donor_db)
+        print ('''
+        Thank you %s for your donation of $%i to this organization. 
+        We really appreciate your generousity and hope to see you again at our upcoming events.
+        Sincerely Yours
+        James Dot
+        ''' % (choose, donation))
+
+
+
+
+def add_donor_db(choose):
+    new_donor = input("Do you want to add %s to the donation list? yes or no " % choose).title()
+    if new_donor == "Yes":
+        amount = input("Enter an amount to donate for %s " % choose)
+        donor_db.update({new_donor:amount})
+        print (donor_db)
+        print ('''
+            Thank you %s for your donation of $%i to this organizat1ion. We really appreciate your generousity and hope to see you again at our upcoming events.
+            Sincerely Yours
+            James Dot
+            ''' % (new_donor, int(amount)))
+
+
+
+
+
+def user_Report(donor_db):
+    print('Donor Name  | ',  'Total Given | ', 'Num Gift  | ', 'Average Gift')
+    for name, donations in donor_db.items():
+        num_of_gift = len(donations)
+        total_given = int(sum(donations))
+        average_gift = int(total_given/num_of_gift)
+        return('{0:<15}   {1:^10}     {2:^6}     {3:8}'.format(name, total_given, num_of_gift, average_gift))
```

Comment:
Placing `return` inside the `for` loop will cause the loop to terminate immediately after the first iteration, and that'll not create the entire report for all donors. Is this what you intended? If you intended this function to create a report for only one donor, why create a loop in the first place? You need to think through your intention and logic here and clarify/fix this code.